examples/pytorch/rgcn/entity_sample_multi_gpu.py

```diff
@@ -1,140 +1,159 @@
-        master_ip='127.0.0.1', master_port='12345')
-    backend = 'nccl'
-    if proc_id == 0:
-        print("backend using {}".format(backend))
-    th.distributed.init_process_group(backend=backend,
-                                      init_method=dist_init_method,
-                                      world_size=n_gpus,
-                                      rank=proc_id)
-
-    device = th.device(dev_id)
-    use_ddp = True if n_gpus > 1 else False
-    train_loader, val_loader, test_loader = init_dataloaders(
-        args, g, train_idx, test_idx, target_idx, dev_id, use_ddp=use_ddp)
-
-    model = RGCN(g.num_nodes(),
-                 args.n_hidden,
-                 num_classes,
-                 num_rels,
-                 num_bases=args.n_bases,
-                 dropout=args.dropout,
-                 self_loop=args.use_self_loop,
-                 ns_mode=True)
+    with torch.no_grad():
+        for input_nodes, output_nodes, blocks in dataloader:
+            output_nodes = inv_target[output_nodes]
+            for block in blocks:
+                block.edata['norm'] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            eval_logits.append(logits.cpu().detach())
+            eval_seeds.append(output_nodes.cpu().detach())
+    eval_logits = torch.cat(eval_logits)
+    eval_seeds = torch.cat(eval_seeds)
+    num_seeds = len(eval_seeds)
+    loc_sum = accuracy(eval_logits.argmax(dim=1), labels[eval_seeds].cpu()) * float(num_seeds)
+    return torch.tensor([loc_sum.item(), float(num_seeds)])
+
+def train(proc_id, device, g, target_idx, labels, train_idx, inv_target, model):
+    # define loss function and optimizer
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=1e-2, weight_decay=5e-4)
+    # construct sampler and dataloader
+    sampler = MultiLayerNeighborSampler([4, 4])
+    train_loader = DataLoader(g, target_idx[train_idx], sampler, device=device, 
+                              batch_size=100, shuffle=True, use_ddp=True)
+    # no separate validation subset, use train index instead for validation
+    val_loader = DataLoader(g, target_idx[train_idx], sampler, device=device, 
+                            batch_size=100, shuffle=False, use_ddp=True)
+    for epoch in range(50):
+        model.train()
+        total_loss = 0
+        for it, (input_nodes, output_nodes, blocks) in enumerate(train_loader):
+            output_nodes = inv_target[output_nodes]
+            for block in blocks:
+                block.edata['norm'] = dgl.norm_by_dst(block).unsqueeze(1)
+            logits = model(blocks)
+            loss = loss_fcn(logits, labels[output_nodes])
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        # torchmetric accuracy defined as num_of_correct_labels / local_num_of_seeds
+        # loc_acc_split returns [local_num_of_seeds, num_of_correct_labels]
```

Comment:
How about `loc_acc_split = (accuracy * batch_size, batch_size)`?

Reply:
Right... I updated it as:
```
# torchmetric accuracy defined as num_of_correct_labels / num_of_nodes
# loc_acc_split = [loc_accuracy * loc_num_of_nodes, loc_num_of_nodes]
```